File: BackEnd/Models/Voo.cs
using System.ComponentModel.DataAnnotations;

public class Voo
{
    [Key]
    public int Id { get; set; }
    public string? Numero { get; set; }
    public Aviao? Aviao { get; set; }
    public int AviaoId { get; set; }
    public Aeroporto? OrigemAeroporto { get; set; }
    public int OrigemAeroportoId { get; set; }
    public Aeroporto? DestinoAeroporto { get; set; }
    public int DestinoAeroportoId { get; set; }
}

File: FrontEnd/AppPassagem/src/app/models/Voo.ts
import { Aeroporto } from "./Aeroporto";
import { Aviao } from "./Aviao";

export class Voo {
  id: number = 0;
  numero: string = '';
  aviao: Aviao;
  aviaoId: number = 0;
  origemAeroporto: Aeroporto;
  origemAeroportoId: number = 0;
  destinoAeroporto: Aeroporto;
  destinoAeroportoId: number = 0;

  constructor(
    numero: string,
    aviao: Aviao,
    aviaoId: number,
    origemAeroportoId: number,
    origemAeroporto: Aeroporto,
    destinoAeroportoId: number,
    destinoAeroporto: Aeroporto
  ) {
    this.numero = numero;
    this.aviao = aviao;
    this.aviaoId = aviaoId;
    this.origemAeroporto = origemAeroporto;
    this.origemAeroportoId = origemAeroportoId;
    this.destinoAeroporto = destinoAeroporto;
    this.destinoAeroportoId = destinoAeroportoId;
  }
}